### ISSUE TITLE
[C++] Cannot enable CUDA on Ubuntu 22.04

### ISSUE THREAD

**Issue by pitrou** [apache/arrow#36870]
### Describe the bug, including details regarding any error messages, version, and platform.

Because of https://gitlab.kitware.com/cmake/cmake/-/issues/25127 , it is difficult or impossible to enable CUDA with Ubuntu 22.04 and the system package for the CUDA toolkit.

However, since we only seem to use `libcuda.so` and `cuda.h` (which is a standalone include file), perhaps we don't need to rely on CMake's `FindCUDAToolkit` at all.

### Component(s)

C++, GPU

**Comment by kou** [comment]
It seems that this is fixed. The upstream issue was closed.

**Comment by pitrou** [comment]
It's not fixed in CMake, as CMake considers one should use the conda packages for CUDA when the conda toolchain is used. I was trying to use the Ubuntu-provided CUDA toolkit inside a conda environment.

**Comment by pitrou** [comment]
That said, we can also choose not to support this, as it might later dysfunction for other reasons.

**Comment by kou** [comment]
Does `-DCUDAToolkit_ROOT=/usr` for you?
I'm using it on Debian GNU/Linux sid.

**Comment by pitrou** [comment]
No, it doesn't. `CUDAToolkit_ROOT` expects a layout that is not the one used by the CUDA system package.

**Comment by kou** [comment]
Hmm. Did you try `-DCUDAToolkit_ROOT=/usr` not `-DCUDAToolkit_ROOT=/usr/lib/x86_64-linux-gnu/`?
How about `-DCUDA_CUDART=/usr/lib/x86_64-linux-gnu/libcudart.so`?
